lib/npg_qc/autoqc/checks/pulldown_metrics.pm

```diff
@@ -56,28 +57,51 @@ has 'max_java_heap_size' => (
     default => $MAX_JAVA_HEAP_SIZE,
 );
 
-has 'picard_jar_path' => (
+has 'picard_module' => (
     is      => 'ro',
-    isa     => 'NpgCommonResolvedPathJarFile',
-    coerce  => 1,
-    default => $PICARD_JAR_NAME,
+    isa     => 'Str',
+    default => $PICARD_COMMAND,
 );
 
-has 'picard_command' => (
-    is         => 'ro',
-    isa        => 'Str',
+# TODO: This needs to be factored out of here, insert_size and alignment_filter_metrics
+has 'reference' => (
+    is => 'ro',
+    isa => 'Str',
     lazy_build => 1,
 );
 
-sub _build_picard_command {
+sub _build_reference {
     my $self = shift;
-    my $command = $self->java_cmd . sprintf q[ -Xmx%s -jar %s VALIDATION_STRINGENCY=SILENT BAIT_INTERVALS=%s TARGET_INTERVALS=%s INPUT=%s OUTPUT=/dev/stdout],
-        $self->max_java_heap_size,
-        $self->picard_jar_path,
-        $self->bait_intervals_path,
-        $self->target_intervals_path,
-        $self->input_files->[0];
-    return $command;
+    my $ref;
+    try {
+        $ref = pop @{$self->refs()};
+    }
+    catch {
+        $self->result->add_comment(
+            q[Error: binary reference cannot be retrieved; cannot run insert size check. ] . $_
```

Comment:
This comment is incorrect. We are retrieving a path to the FASTA reference, not to the binary index. The name of the check is wrong here. Failure to retrieve a reference is an error. Not sure whether try-catch block is needed.

Reply:
Do you want a failed call to $self->refs() to kill the invocation of `qc`? I'm unclear on when code needs to be tolerant and when it needs to fail.

Reply:
The reference is needed for gatk, so there is no point proceeding without one. BTW, multiple references can be returned, this should be considered as an error since we would not know which one to use.

Reply:
Checking references seems like the sort of thing you might check in `can_run` instead. Would that be a good idea?

Reply:
It's often done in `can_run`. Sometimes we legitimately do not have a reference and that's OK, we will not run the check then. Pulldowns are only done for a limited number of species. The bait and interval files are linked to a particular reference. We request aligned input files. It would be an error not to have the reference at run time. I hope the code for finding the bait and interval files will error in case of these files missing before we get to the point of invoking gatk